### PR TITLE
Add Product listing page

### PR DESCRIPTION
**Pull request by varun13169**
Add Product listing page
	- grid layout used to generate holy grail.
	- use component lib to layout the components.

### CODE REVIEW

**Review by zoef-yt — APPROVED** [review]
Good to merge

File: pages/productListingPage/productListingPage.css [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+:root {}
+
+body {}
+
+/* main content */
+.main-content {
+    width: 100%;
+    height: 100%;
+    display: grid;
```
Comment (dev-prabhat) [inline pages/productListingPage/productListingPage.css:11]: put width:100% and height: 100% in single class and use

File: siteComponents/sidebar.css [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    flex-direction: column;
+}
+
+.sidebar__fliter {
+    display: flex;
+    margin-bottom: 2.4rem;
+}
+
+.sidebar__fliter button {
+    margin-left: auto;
```
Comment (dev-prabhat) [inline siteComponents/sidebar.css:16]: avoid using tag to apply css

**Review by dev-prabhat — CHANGES_REQUESTED** [review]

File: siteComponents/sidebar.css (hunk 2 above)
Comment (varun13169) [reply siteComponents/sidebar.css:16]: > .sidebar__fliter button {

tag is used for a specific element inside the sidebar.

File: pages/productListingPage/productListingPage.css (hunk 1 above)
Comment (varun13169) [reply pages/productListingPage/productListingPage.css:11]: .main-content  will always be 100% x 100%

Above statement is specific to the project structure that I am following. Following is being done to decouple grid section with the main "main" section if decoupling was not done then that 100% x 100% would not have been required.

**Review by dev-prabhat — APPROVED** [review]

**Pull request merged**